src/main/java/uk/gov/pay/products/service/PaymentCreator.java

```diff
@@ -43,24 +45,24 @@ public PaymentCreator(Provider<TransactionFlow> transactionFlowProvider, Product
         this.linksDecorator = linksDecorator;
     }
 
-    public Payment doCreate(Integer productId) {
+    public Payment doCreate(String productExternalId) {
         PaymentEntity paymentEntity = transactionFlowProvider.get()
-                .executeNext(beforePaymentCreation(productId))
+                .executeNext(beforePaymentCreation(productExternalId))
                 .executeNext(paymentCreation())
                 .executeNext(afterPaymentCreation())
                 .complete().get(PaymentEntity.class);
 
         if (paymentEntity.getStatus() == PaymentStatus.ERROR) {
-            throw new PaymentCreatorDownstreamException(paymentEntity.getProductEntity().getId());
+            throw new PaymentCreatorDownstreamException(paymentEntity.getProductEntity().getExternalId());
         }
         return linksDecorator.decorate(paymentEntity.toPayment());
     }
 
-    private TransactionalOperation<TransactionContext, PaymentEntity> beforePaymentCreation(Integer productId) {
+    private TransactionalOperation<TransactionContext, PaymentEntity> beforePaymentCreation(String productExternalId) {
         return context -> {
-            logger.info("Creating a new payment for productId {}", productId);
-            ProductEntity productEntity = productDao.findById(productId)
-                    .orElseThrow(() -> new PaymentCreatorNotFoundException(productId));
+            logger.info("Creating a new payment for product external id {}", productExternalId);
+            ProductEntity productEntity = productDao.findByExternalId(productExternalId)
+                    .orElseThrow(() -> new PaymentCreatorNotFoundException(productExternalId));
```

Comment:
This is a `ProductNotFoundException` as far as I'm concerned.

Reply:
See my comments above.